src/libraries/Directory.Build.targets

```diff
@@ -356,4 +368,62 @@
            Text="Analyzers must only target netstandard2.0 since they run in the compiler which targets netstandard2.0. The following files were found to target '%(_AnalyzerPackFile.TargetFramework)': @(_AnalyzerPackFile)" />
   </Target>
 
+  <Target Name="AddRefAssemblies" Condition="'$(ExcludeFromPackage)' == '' and '$(TargetsAnyOS)' == 'true'"> 
+    <PropertyGroup>
+      <RefPath>$(BaseOutputPath)/ref/$(TargetFrameworkWithoutSuffix)-$(Configuration)/$(TargetName).dll</RefPath>
```

Comment:
This hardcoded paths make me a little bit nervous... is there a way we could get all these paths differently, i.e calling GetTargetPath or something?

Reply:
i was able to leverage the existing targets for runtime item as those assemblies are built by the project. unfortunately there is no good way to get the ref assemblies path.

Reply:
> unfortunately there is no good way to get the ref assemblies path.

We could mimic how we get it for `GenerateReferenceSource` target right?

Reply:
> unfortunately there is no good way to get the ref assemblies path.

In case we still need refs in our packages why can't you use
`@(ResolvedMatchingContract)`: https://github.com/dotnet/runtime/blob/3924d0308110f6365e47e784b8c7cf1e8f7efa2b/eng/resolveContract.targets#L18?

Reply:
Agree with @ViktorHofer & @safern here as I mentioned in https://github.com/dotnet/runtime/pull/54250#discussion_r658886509.  Leverage the existing target if we can.  Could add metadata to it if necessary.  Be careful about duplicates (multiple src TFMs resulting in same ref TFM).

Reply:
I was able to use ResolvedMatchingContract, i had to add some transformations to remove the duplicates (as i didnt wanted to disable the warning for multiple files), i tested it by adding a net5.0 target manually to the src project.

Reply:
@ericstj can you please take a look at the "transformation to remove the duplicates" change that @Anipik applied? I'm not yet sold on if there isn't a better way to accomplish the task.